### ISSUE TITLE
When I keep scrolling while things are loading into the timeline, the view eratically jumps up which is disorienting

### ISSUE THREAD

**Issue by ell1e** [mastodon/mastodon#21918]
### Steps to reproduce the problem

1. Go to e.g. "federated" view
2. Gently keep scrolling down with mouse wheel
3. View keeps jumping up even though you're scrolling down

### Expected behaviour

No weird jumps of the view on its own

### Actual behaviour

Erratic jumps upward which is disorienting

### Detailed description

Any upward movement you see in this screen cast is 100% involuntary and not triggered by me:
[Screencast from 12-01-2022 09:13:10 AM.webm](https://user-images.githubusercontent.com/64124388/205000817-5e339dd3-27da-44b7-aa2d-0b2c424fb925.webm)

### Specifications

I see this on mastodon.social with Firefox 108 on Linux

**Comment by rbairwell** [comment]
This happens if new posts are coming in (which, on the federated timeline now is very often!) - on local or home timelines you probably won't have this issue (and, tbh, I haven't looked at the federated one for ages as it's just "too general" for me).

If you do find it annoying, go into preferences and select "Slow mode": this means that posts won't autoload into your timeline without you clicking the "Load more" option at the top (when available). I have Slow Mode enabled anyway for mental health/avoiding "doomscrolling" reasons.

**Comment by ell1e** [comment]
Slow mode seems to resolve it. I think slow mode should be the default, Twitter I think also requires a click to show new tweets that were posted in the meantime and this viewport jumping as a default behavior is just very disorienting (and makes Mastodon look bad compared to Twitter).

**Comment by ell1e** [comment]
Isn't there any way in JavaScript to just avoid the wild jumping around if something is inserted above? I think this should be investigated. Because this seems to be *the* main downside of not defaulting to slow mode, if it could be fixed it'd be much less of an issue.

Alternatively, I'm wondering if non-slow mode should just be removed entirely. Twitter doesn't seem to offer anything similar, and the drawback of the disorienting jumps doesn't seem to be worth any potential benefits of spared clicks to see new content.

**Comment by YourAutisticLife** [comment]
@rbairwell 

No, going into slow mode does not actually fix it. It will look like it works for a while, but eventually the code will slowly lose track of the proper position to keep, and things will start to go sideways.

I have a PR that fixes it:

https://github.com/mastodon/mastodon/pull/26380
